For immunogram to be the general framework to measure a person's immune system, it must have one-for-all robustness. Meaning it must be able to explain not only immune phenotypes but also cell types. Furthermore, immunogram that can differentiate clinical phenotypes without an ability to differentiate immune cell types seems implausible, because immune cell types are an important factor of immune phenotypes. One-for-all robustness requires immunogram to detect various related phenotypes, including cell types.

So, if immune interactions are indeed the generalized truth (equivalent to organs) of the immune system, then their interactions alone should capture cell-type information. If they are not, then looking at their interactions alone should be less informative in identifying cell types compared to looking at the entire transcriptome. I tested this hypothesis using the data from https://www.sciencedirect.com/science/article/pii/S2211124719300592?via%3Dihub.

Gianni et al samples blood from people and used cell surface markers to isolate the blood cells. For each group of isolate cells, they did RNA sequencing. Their data set provides cell type labels and corresponding gene expression data.

![Gianni et al's process in generating the cell type data](https://github.com/KwatMDPhD/ImmunePopulation.pro/raw/main/input/rna_seq_signatures_normalized_by_mrna_abundance_allow_absolute_deconvolution_of_human_immune_cell_types/5.1-s2.0-S2211124719300592-gr1_lrg.jpg)

[Gianni et al's surface-marker based labels and corresponding samples' transcriptome](https://github.com/KwatMDPhD/ImmunePopulation.pro/raw/main/output/GSE107011/gene_x_celltype_x_tpm+1log.html)

There are 29 cell types. So I clustered the transcriptome data into 2 to 29 groups, and scored the ability of the data to capture the cell types by the percentage of times the samples from a same cell type grouped together. A score of 0 means that from clustering the samples into 2 groups to 29 groups, no cell type had its samples grouped. On the other hand, a score of 100 means that during each clustering, from 2 to 29, every cell type had its sampled grouped.

[Clustering Gianni et al's transcriptome data into 2 to 29 groups results in the samples of a cell type being grouped together 69.33% of the times](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/Gene.cluster.group.html)

Clustering the transcriptome consistently grouped some cell types more than others. The ones whose samples grouped together more were cell types with more committed lineage like neutrophils. In contrast, the ones whose samples poorly grouped were less differentiated cell types like progenitors.

[Clustering Gianni et al's transcriptome data results in some cell types being co-grouped more than others](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/Gene.cluster.tight.html)

## Can elements of a simple immune interaction encode the immune cell types?

![example 1 frame](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/frame.png)

![example 1 DZQV_CD4_naive heat](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/DZQV_CD4_naive/1.png)

[example 1 heat matrix](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/Heated.html)

[example 1 heat matrix cluster group](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/Heated.cluster.group.html)

[example 1 heat matrix cluster tight](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/Heated.cluster.tight.html)

[example 1 heat matrix cluster change](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/Heated.cluster.change.html)

Yes. Elements from a simple immune interaction can encode their relevant cell types.

## Does running the interaction improve the encoding?

![example 1 DZQV_CD4_naive flow](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/DZQV_CD4_naive/animate.gif)

[example 1 flow matrix](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/Flowed.html)

[example 1 flow matrix cluster group](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/Flowed.cluster.group.html)

[example 1 flow matrix cluster tight](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/Flowed.cluster.tight.html)

[example 1 flow matrix cluster change](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.1.no_fill/Flowed.cluster.change.html)

Yes. Running interactions provides additional information that improves the encoding.

## Does increasing the interaction improve the encoding?

![example 2 frame](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/frame.png)

![example 2 DZQV_CD4_naive heat](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/DZQV_CD4_naive/1.png)

[example 2 heat matrix](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/Heated.html)

[example 2 heat matrix cluster group](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/Heated.cluster.group.html)

[example 2 heat matrix cluster tight](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/Heated.cluster.tight.html)

[example 2 heat matrix cluster change](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/Heated.cluster.change.html)

![example 2 DZQV_CD4_naive flow](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/DZQV_CD4_naive/animate.gif)

[example 2 flow matrix](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/Flowed.html)

[example 2 flow matrix cluster group](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/Flowed.cluster.group.html)

[example 2 flow matrix cluster tight](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/Flowed.cluster.tight.html)

[example 2 flow matrix cluster change](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.no_fill/Flowed.cluster.change.html)

Yes. More potential interactions improves the encoding.

## Does simulation improve the encoding?

![example 2 fill DZQV_CD4_naive heat](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.fill_antigen_and_cells/DZQV_CD4_naive/1.png)

[example 2 fill heat matrix](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.fill_antigen_and_cells/Heated.html)

![example 2 fill DZQV_CD4_naive flow](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.fill_antigen_and_cells/DZQV_CD4_naive/animate.gif)

[example 2 fill flow matrix](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.fill_antigen_and_cells/Flowed.html)

[example 2 fill flow matrix cluster group](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.fill_antigen_and_cells/Flowed.cluster.group.html)

[example 2 fill flow matrix cluster tight](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.fill_antigen_and_cells/Flowed.cluster.tight.html)

[example 2 fill flow matrix cluster change](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.2.fill_antigen_and_cells/Flowed.cluster.change.html)

Yes. Simulation enables more interactions and improves the encoding.

## How much information is encoded in a classical immune interaction?

![example 3 fill frame](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.3.fill_antigen_and_cells/frame.png)

![example 3 fill DZQV_CD4_naive flow](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.3.fill_antigen_and_cells/DZQV_CD4_naive/animate.gif)

[example 3 fill flow matrix](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.3.fill_antigen_and_cells/Flowed.html)

[example 3 fill flow matrix cluster group](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.3.fill_antigen_and_cells/Flowed.cluster.group.html)

[example 3 fill flow matrix cluster tight](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.3.fill_antigen_and_cells/Flowed.cluster.tight.html)

[example 3 fill flow matrix cluster change](https://github.com/KwatMDPhD/MolecularMedicine.pro/raw/main/output/cell_20220919/example.3.fill_antigen_and_cells/Flowed.cluster.change.html)

Classical immune interactions may encode as much cell-type information as the transcriptome.
